doc/specifications/translations/removal.md

```diff
@@ -14,7 +14,7 @@ the existing Versions of a Content Object and is specified as:
  *       a warning before performing it.
  *
  * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException if the specified translation
- *         is the only one Content Object has.
+ *         is the only one Content Object has or it is the main translation.
```

Comment:
> for the last one on exceptions, afaik we have:
> - main
> - draft with language _(as it would imply removing the draft)_
> - just one translation

I think if we have "just one translation" it also implies "main", right?

As it goes for draft with language:
- AFAICS if we create draft from version with multiple translations (by calling `ContentService::createContentDraft`), these translations will be copied, so won't match the case with only one language.
- if we create a draft from version which has only one translation and later on try to remove the this translation, we'll get `BadStateException`. But that goes also for the source version for that draft.
- actually, there's no distinction between draft, archived and published. The check is for sanity of removal of the translation - if it's the only one, it results in `BadStateException`. I think in any case removeTranslation *should not* result in complete removal of a version and this is what this implementation tries to protect user from. 
It is achieved by 1) First doing validation on Core ContentService level, and then 2) checking result of update of language mask on gateway level and rolling back if an update resulted in the mask `b'0'` or `b'1'` (alwaysAvailable).